### PR TITLE
Add development setup instructions for Archlinux

### PR DESCRIPTION
**Pull request by erikthered**

### CODE REVIEW

**Comment by it33** [comment]
Thanks @erikthered! Much appreciated! 

Open question to the community: Can anyone who runs Arch please try out these instructions and add +1 if they work for you, or suggest any feedback?

**Comment by ninja-** [comment]
this works for me although I remember having some problems with Makefile, I think it's important to run it exactly via
```
make build run
```

**Comment by erikthered** [comment]
I don't recall having any problems using just `make run`, maybe someone else can chime in?

**Comment by ninja-** [comment]
@erikthered that doesn't seem to recompile the server for me...

**Comment by ninja-** [comment]
I can' get the server to compile easily...

make build-server
```bash
gofmt sucess
cp ./model/version.go ./model/version.go.bak
sed -i'.make_mac_work' 's|_BUILD_NUMBER_|dev|g' ./model/version.go
sed -i'.make_mac_work' 's|_BUILD_DATE_|...|g' ./model/version.go
sed -i'.make_mac_work' 's|_BUILD_HASH_|14d44a2ce0fe7ea2708df81d6ad2e98e7ea5d340|g' ./model/version.go
rm ./model/version.go.make_mac_work
/home/ninja/go/bin/godep go build  ./...
package ./src/github.com/garyburd/redigo/redis
        imports github.com/garyburd/redigo/internal: use of internal package not allowed
package ./src/github.com/garyburd/redigo/redisx
        imports github.com/garyburd/redigo/internal: use of internal package not allowed
src/github.com/goamz/goamz/cloudwatch/cloudwatch.go:22:2: cannot find package 
```

and repeats for every package...my GOPATH is 100% fine according to docs. does it work for you @erikthered ?

**Comment by erikthered** [comment]
I didn't even try `make build-server`, let me give that a try and get back to you.

**Comment by ninja-** [comment]
another try after cleaning up results in this...
```
# _/home/ninja/platform/utils
utils/config.go:208: c.SupportSettings undefined (type *model.Config has no field or method SupportSettings)
utils/config.go:209: c.SupportSettings undefined (type *model.Config has no field or method SupportSettings)
utils/config.go:210: c.SupportSettings undefined (type *model.Config has no field or method SupportSettings)
utils/config.go:211: c.SupportSettings undefined (type *model.Config has no field or method SupportSettings)
utils/config.go:212: c.SupportSettings undefined (type *model.Config has no field or method SupportSettings)
utils/config.go:213: c.SupportSettings undefined (type *model.Config has no field or method SupportSettings)
# _/home/ninja/platform/store
store/sql_channel_store.go:609: utils.Cfg.GetSanitizeOptions undefined (type *model.Config has no field or method GetSanitizeOptions)
5848294 bytes written to ../static/js/bundle.js (7.28 seconds)
# _/home/ninja/platform/api
api/channel.go:307: undefined: model.POST_HEADER_CHANGE
api/post.go:276: undefined: model.ACTION_PREFERENCE_CHANGED
api/post.go:291: undefined: model.ACTION_PREFERENCE_CHANGED
api/user.go:664: utils.Cfg.GetSanitizeOptions undefined (type *model.Config has no field or method GetSanitizeOptions)
api/user.go:1105: utils.Cfg.GetSanitizeOptions undefined (type *model.Config has no field or method GetSanitizeOptions)
api/user.go:1225: utils.Cfg.GetSanitizeOptions undefined (type *model.Config has no field or method GetSanitizeOptions)
api/user.go:1551: utils.Cfg.GetSanitizeOptions undefined (type *model.Config has no field or method GetSanitizeOptions)
api/web_team_hub.go:101: undefined: model.ACTION_PREFERENCE_CHANGED
```

**Comment by ninja-** [comment]
ok I can't read properly...the doc said to have the platform folder in goroot/src/github.com/mattermost. finally got it working

**Comment by crspeller** [comment]
Hi @erikthered  `make build-server` is an internal target and really shouldn't be called directly. `make run` works fine. If you want to create a build like the build server turns out you can run `make dist-local`. 

Taking a look at your instructions now.

File: doc/developer/Setup.md [hunk 1]
```diff
@@ -81,3 +81,51 @@ Any issues? Please let us know on our forums at: http://forum.mattermost.org
...
+	1. Edit your /etc/hosts file to include the following line
+		`127.0.0.1 dockerhost`
+3. Install Go
+	1. `pacman -S go`
+4. Set up your Go workspace and add Go to the PATH
+	1. `mkdir ~/go`
+	2. Add the following to your ~/.bashrc
+		1. `export GOPATH=$HOME/go`
+		2. `export GOROOT=/usr/lib/go`
+		3. `export PATH=$PATH:$GOROOT/bin`
```
Comment (crspeller) [inline doc/developer/Setup.md]: For me it sets GOROOT automatically when I run `pacman -S go`
Also I think the PATH should be `PATH=$PATH:$GOPATH/bin` as that's the one pacman doesn't setup for you.

**Comment by erikthered** [comment]
@crspeller yeah you're totally right, the PATH line is wrong. I'll update it and remove the GOROOT line.

**Comment by crspeller** [comment]
Looks good. Thanks @erikthered!

**Pull request merged**